packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -271,116 +268,114 @@ class RetainingPathWidget extends StatelessWidget {
-  );
-  if (retainingPath.elements == null) return [emptyList];
-
-  final firstRetainingObject = retainingPath.elements!.isNotEmpty
-      ? SelectableText(
-          _objectName(retainingPath.elements!.first.value) ??
-              '<RetainingObject>',
-          style: Theme.of(context).fixedFontStyle,
-        )
-      : emptyList;
-
-  final retainingObjects = [
-    Row(children: [firstRetainingObject]),
-    if (retainingPath.elements!.length > 1)
-      for (RetainingObject object in retainingPath.elements!.sublist(1))
-        Row(
-          children: [
-            SelectableText(
-              _retainingObjectDescription(object),
-              style: Theme.of(context).fixedFontStyle,
-            ),
-          ],
-        ),
-    Row(
-      children: [
-        SelectableText(
-          'Retained by a GC root of type ${retainingPath.gcRootType ?? '<unknown>'}',
-          style: Theme.of(context).fixedFontStyle,
-        ),
-      ],
-    )
-  ];
+  /// Returns a list of Widgets that will be the rows in the VmExpansionTile
+  /// for RetainingPathWidget.
+  List<Widget> _retainingPathList(
+    BuildContext context,
+    RetainingPath retainingPath,
+  ) {
+    final emptyList = SelectableText(
+      'No retaining objects',
+      style: Theme.of(context).fixedFontStyle,
+    );
+    if (retainingPath.elements == null) return [emptyList];
 
-  final retainingObjectsRows = _prettyRows(context, retainingObjects);
+    final firstRetainingObject = retainingPath.elements!.isNotEmpty
+        ? SelectableText(
+            _objectName(retainingPath.elements!.first.value) ??
+                '<RetainingObject>',
+            style: Theme.of(context).fixedFontStyle,
+          )
+        : emptyList;
+
+    final retainingObjects = [
+      Row(children: [firstRetainingObject]),
+      if (retainingPath.elements!.length > 1)
+        for (RetainingObject object in retainingPath.elements!.sublist(1))
+          Row(
+            children: [
+              SelectableText(
+                _retainingObjectDescription(object),
+                style: Theme.of(context).fixedFontStyle,
+              ),
+            ],
+          ),
+      Row(
+        children: [
+          SelectableText(
+            'Retained by a GC root of type ${retainingPath.gcRootType ?? '<unknown>'}',
+            style: Theme.of(context).fixedFontStyle,
+          ),
+        ],
+      )
+    ];
 
-  return <Widget>[...retainingObjectsRows];
-}
+    final retainingObjectsRows = _prettyRows(context, retainingObjects);
 
-/// Describes the given RetainingObject [object] and its parentListIndex,
-/// parentMapKey, and parentField where applicable.
-String _retainingObjectDescription(RetainingObject object) {
-  if (object.parentListIndex != null) {
-    final ref = object.value as InstanceRef;
-    return 'Retained by element [${object.parentListIndex}] of ${ref.classRef?.name ?? '<parentListName>'}';
+    return <Widget>[...retainingObjectsRows];
```

Comment:
Is there any reason we're returning a new list here instead of just returning `retainingObjectsRows`?

Reply:
Yeah... I had done some refactoring and then it stayed that way. Thanks for telling me!